tests/test_precommit_setup.py

```
@@ -0,0 +1,197 @@
+        )
+
+    def test_precommit_in_dev_dependencies(self):
+        """Test that pre-commit is listed in development dependencies."""
+        # Arrange
+        repo_root = Path(__file__).parent.parent
+        pyproject_file = repo_root / "pyproject.toml"
+
+        # Act & Assert
+        assert pyproject_file.exists(), "pyproject.toml should exist"
+
+        with open(pyproject_file) as f:
```

Ruff (0.12.2)

`open()` should be replaced by `Path.open()`

(PTH123)